app/models/flexible_criterion.rb

```diff
@@ -100,6 +100,16 @@ def get_weight
     1
   end
 
+  # Returns the value entered when the criterion is set up.
+  def mark_set_up_value
+    max_mark
+  end
+
+  # Returns the maximum mark for a particular criterion.
+  def max_mark
```

Comment:
The first situation you described is not a good reason to use a different name. Please make them the same.